py_ecc/fields/optimized_field_elements.py

```diff
@@ -30,6 +30,15 @@
 IntOrFQ = Union[int, T_FQ]
 
 
+def mod_int(x: IntOrFQ, n: int) -> int:
+    if isinstance(x, int):
+        return x % n
+    elif isinstance(x, FQ):
+        return x.n % n
```

Comment:
Instead of special casing this, what do you think about implementing `FQ.__mod__`

https://docs.python.org/3/reference/datamodel.html#object.__mod__

Reply:
I was thinking about it, but there is a stub says `FQ.__mod__` is for `IntOrFQ`modulus:
```
def __mod__(self: T_FQ, other: IntOrFQ) -> T_FQ:
        raise NotImplementedError("Modulo Operation not yet supported by fields")
```
Having `mod_int` might be more explicit of the usage by its name.